src/helpers/3box.ts

```diff
@@ -11,4 +11,47 @@ export async function getProfiles(addresses) {
       image: true
     }
   });
+
+  const ensProfiles = await subgraphRequest(
+    'https://api.thegraph.com/subgraphs/name/ensdomains/ens',
+    {
+      accounts: {
+        __args: {
+          first: 1000,
+          where: {
+            id_in: addresses.map(address => address.toLowerCase())
+          }
+        },
+        id: true,
+        domains: {
+          __args: {
+            first: 1
+          },
+          name: true
+        }
+      }
+    }
+  );
+
+  ensProfiles.accounts.forEach(account => {
+    if (account.domains && account.domains.length > 0 && account.id) {
+      const profileFoundIndex = profiles.profiles.findIndex(
+        profile =>
+          profile.eth_address.toLowerCase() === account.id.toLowerCase()
+      );
+      if (profileFoundIndex > -1) {
+        if (!profiles.profiles[profileFoundIndex].name) {
+          profiles.profiles[profileFoundIndex].name = account.domains[0].name;
+        }
+      } else {
+        profiles.profiles.push({
+          name: account.domains[0].name,
+          eth_address: account.id,
+          ens: true
+        });
+      }
+    }
+  });
+
+  return profiles;

```

Comment:
I think this method should return an object with the addresses as key and profile as value like this:
```
{
  '0x123456': {
    name: "Fabien", // 3box name or ENS domain
    ens: 'fabien.eth'
    image: 'https://...'
  }
  '0x235': {
    name: 'Bob'
    ...
  },
  '0x456': {} // If empty profile
}
```
That will make the data easier to consume with this format. Also we don't need to return `eth_address`.